### PR TITLE
add tests

### PR DESCRIPTION
**Pull request by instabledesign**

### CODE REVIEW

**Comment by rmasclef** [comment]
OK tests passes 👍 

We will need to work on the `TestNewTimeRotateFileSream_Handle` in order to make it pass too

File: context.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func (c *Context) stringTo(builder *strings.Builder) *Context {
 
 // String will return context as string
 func (c *Context) String() string {
-	if len(*c) == 0 {
+	if c == nil || len(*c) == 0 {
```
Comment (rmasclef) [inline context.go]: Can you confirm we need to add this case ?

File: handler/rotate_test.go [hunk 2]
```diff
@@ -7,35 +7,45 @@ import (
...
+
 	"github.com/gol4ng/logger"
 	"github.com/gol4ng/logger/handler"
 	"github.com/gol4ng/logger/mocks"
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/mock"
 )
 
 func TestNewTimeRotateFileStream_Handle(t *testing.T) {
+	// t.Skip("rework this test")
```
Comment (rmasclef) [inline handler/rotate_test.go]: remove that

File: handler/rotate_test.go [hunk 3]
```diff
@@ -7,35 +7,45 @@ import (
...
-
 		assert.True(t, false, "must not be reached")
 		return 0, nil
 	})
-
+	// mock os.File::Close method in order not to return an error
+	// on the second call of rotate, the TimeFileProvider will have a writer that is not nil (it will contain `createdFile1` that is technically nil
+	// if we do not mock the `Close` method on `createdFile1` which will and up with a syscall.EINVAL error
+	// as the code will pass here https://github.com/golang/go/blob/release-branch.go1.12/src/os/file_unix.go#L242
+	monkey.PatchInstanceMethod(reflect.TypeOf(f), "Close", func(file *os.File) error {
```
Comment (rmasclef) [inline handler/rotate_test.go]: important part for passing the `rotate_file_test` 👍 
feel free to reach me if you need more explanations 😄

**Comment by rmasclef** [comment]
OK all tests pass now, including `TestNewTimeRotateFileSream_Handle` 🎉

File: context.go (hunk 1 above)
Comment (instabledesign) [reply context.go]: yes 👍 nice one

File: handler/rotate_test.go (hunk 3 above)
Comment (instabledesign) [reply handler/rotate_test.go]: omg yes nice catch!!!

**Pull request merged**